examples_tests/39.DenoiserTonemapper/main.cpp

```diff
@@ -586,152 +584,159 @@ void main()
+	}
+
 
 #define DENOISER_BUFFER_COUNT 3u //had to change to a define cause lambda was complaining about it not being a constant expression when capturing
 	// keep all CUDA links in an array (less code to map/unmap
-	cuda::CCUDAHandler::GraphicsAPIObjLink<video::IGPUBuffer> bufferLinks[DENOISER_BUFFER_COUNT];
+	cuda::CCUDAHandler::GraphicsAPIObjLink<video::IGPUBuffer> bufferLinks[DENOISER_BUFFER_COUNT + 1];
 	// except for the scratch CUDA buffer which can and will be ENORMOUS
 	CUdeviceptr denoiserScratch = 0ull;
 	// set-up denoisers
 	constexpr size_t IntensityValuesSize = sizeof(float);
 	auto& intensityBuffer = bufferLinks[0];
 	auto& denoiserState = bufferLinks[0];
 	auto& temporaryPixelBuffer = bufferLinks[1];
-	auto& imagePixelBuffer = bufferLinks[2];
+	auto& imagePixelBuffer = bufferLinks[2]; //buffer to store result of denoising of a tile/image
 	size_t denoiserStateBufferSize = 0ull;
 	{
 		size_t scratchBufferSize = 0ull;
 		size_t pixelBufferSize = 0ull;
-		for (uint32_t i=0u; i<EII_COUNT; i++)
+		for (uint32_t i = 0u; i < EII_COUNT; i++)
 		{
 			auto& denoiser = denoisers[i].m_denoiser;
-			if (maxResolution[i][0]==0u || maxResolution[i][1]==0u)
+			if (maxResolution[i][0] == 0u || maxResolution[i][1] == 0u)
 			{
 				denoiser = nullptr;
 				continue;
 			}
-
+				
 			OptixDenoiserSizes m_denoiserMemReqs;
-			if (denoiser->computeMemoryResources(&m_denoiserMemReqs, maxResolution[i])!=OPTIX_SUCCESS)
+			if (denoiser->computeMemoryResources(&m_denoiserMemReqs, maxResolution[i]) != OPTIX_SUCCESS)
 			{
-				static const char* errorMsgs[EII_COUNT] = {	"Failed to compute Color-Denoiser Memory Requirements!",
+				static const char* errorMsgs[EII_COUNT] = { "Failed to compute Color-Denoiser Memory Requirements!",
 															"Failed to compute Color-Albedo-Denoiser Memory Requirements!",
-															"Failed to compute Color-Albedo-Normal-Denoiser Memory Requirements!"};
-				os::Printer::log(errorMsgs[i],ELL_ERROR);
+															"Failed to compute Color-Albedo-Normal-Denoiser Memory Requirements!" };
+				os::Printer::log(errorMsgs[i], ELL_ERROR);
 				denoiser = nullptr;
 				continue;
 			}
-
 			denoisers[i].stateOffset = denoiserStateBufferSize;
 			denoiserStateBufferSize += denoisers[i].stateSize = m_denoiserMemReqs.stateSizeInBytes;
-			scratchBufferSize = core::max(scratchBufferSize,denoisers[i].scratchSize = m_denoiserMemReqs.recommendedScratchSizeInBytes);
-			pixelBufferSize = core::max(pixelBufferSize,core::max(asset::getTexelOrBlockBytesize(EF_R32G32B32A32_SFLOAT),(i+1u)*forcedOptiXFormatPixelStride)*maxResolution[i][0]*maxResolution[i][1]);
+			scratchBufferSize = core::max(scratchBufferSize, denoisers[i].scratchSize = m_denoiserMemReqs.withOverlapScratchSizeInBytes);
+			pixelBufferSize = core::max(pixelBufferSize, core::max(asset::getTexelOrBlockBytesize(EF_R32G32B32A32_SFLOAT), (i + 1u) * forcedOptiXFormatPixelStride) * maxResolution[i][0] * maxResolution[i][1]);
```

Comment:
`pixelBufferSize` is important because it gets used for storing the actual image as a whole for autoexposure and luminancce measure, you're gonna have a problem if this buffer is sized for a tile and not a whole image.